src/Sarif/Core/FileData.cs

```diff
@@ -36,13 +36,13 @@ public static IList<FileData> Create(IEnumerable<Uri> uris, out string fileDataK
                 }
                 else if (files.Count == 1)
                 {
-                    fileData.PathFromParent = "#" + uri.ToString();
-                    fileDataKey = fileDataKey + fileData.PathFromParent;
+                    fileData.PathFromParent = uri.ToString();
+                    fileDataKey = fileDataKey + "#" + fileData.PathFromParent;
```

Comment:
You wrote "# characters...", and Markdown interpreted the "#" to mean "first-level heading".